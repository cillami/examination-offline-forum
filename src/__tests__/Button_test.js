import React from "react";
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Button from "../components/Button";


it("test button", () => {
  const fakeClick = jest.fn();
  const buttonDanger = true;
  const wrapper = mount(
    <Button onClick={fakeClick} danger={buttonDanger} >
      <div> children </div>
    </Button>
  );
  expect(wrapper.find("button").hasClass("bg-red-dark")).toEqual(true);
}); 


it('should match snapshot', () => {
  const fakeClick = jest.fn();
  const noDanger = false;
  const wrapper = mount(
    <Button onClick={fakeClick} danger={noDanger} >
      <div> children </div>
    </Button>
  );
  expect(wrapper.find("button").hasClass("bg-red-dark")).toEqual(false);
  expect(toJSON(wrapper)).toMatchSnapshot();
});