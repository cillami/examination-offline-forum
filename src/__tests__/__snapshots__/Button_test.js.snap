// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<Button
  className=""
  danger={false}
  onClick={[Function]}
>
  <button
    className="bg-indigo-dark hover:bg-indigo-darker text-white font-bold py-2 px-4 rounded-full "
    data-test="button"
    onClick={[Function]}
  >
    <div>
       children 
    </div>
  </button>
</Button>
`;
